src/cmd/build.rs

```diff
@@ -430,6 +430,9 @@ pub(crate) fn execute_with_crate_metadata(
         "Building cargo project".bright_green().bold()
     );
     build_cargo_project(&crate_metadata, build_artifact, verbosity, unstable_flags)?;
+    if build_artifact == BuildArtifacts::CheckOnly {
+        return Ok((None, None));
+    }
```

Comment:
This early return is needed because the next step would otherwise try to post-process the contract. But check does not actually create a contract.

Without this change `cargo contract check` fails with 
```
 [2/2] Post processing wasm file
ERROR: Loading original wasm file '/tmp/foobar/target/ink/wasm32-unknown-unknown/release/foobar.wasm'

Caused by:
    Can't read from the file: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```